backend/mqttauth/views.py

```diff
@@ -68,45 +76,49 @@ def acl(request):
     URI-Param           username   password    clientid    topic   acc
     http_aclcheck_uri   Y          N           Y           Y       Y
     """
-    response_status_code = 403
+    response_status_code = status.HTTP_403_FORBIDDEN
 
     username = request.POST.get('username')
     topic = request.POST.get('topic')
     # client_id = request.POST.get('clientid')
     acc = request.POST.get('acc')  # 1 == sub, 2 == pub
 
     permission_name_map = {
-        '1': 'subscribe_datastream',
-        '2': 'publish_datastream',
+        '1': 'view_datastream',
+        '2': 'create_observation',
     }
 
     permission_name = permission_name_map.get(acc)
     if not permission_name:
         logger.info('MQTT ACL check encountered unknown value for acc: "{}"'.format(acc))
         return HttpResponse(status=response_status_code)
 
-    log_text = 'MQTT ACL check for user "{}", topic "{}", perm "{}". Result: '.format(username, topic, permission_name)
+    log_text = 'MQTT ACL check for service "{}", topic "{}", perm "{}". Result: '.format(
+        username, topic, permission_name)
 
-    user = get_user_by_username_or_anonymous(username)
-
-    if user.is_superuser:
-        logger.info(log_text + 'Access granted (superuser)')
-        return HttpResponse(status=200)
+    try:
+        service = Service.objects.get(identifier=username)
+    except Service.DoesNotExist:
+        logger.info(log_text + 'Unknown service.')
+        return HttpResponse(status=response_status_code)
 
     parse_result = parse_sta_url(topic, prefix=settings.STA_VERSION)
 
-    if parse_result['parts']:
+    if parse_result and parse_result['parts']:
         # For now we only look for a Datastream id and check permissions for that datastream
         for part in parse_result['parts']:
             if part['type'] == 'entity' and part['name'] == 'Datastream' and part['id']:
                 try:
                     ds = Datastream.objects.get(sts_id=part['id'])
+                    permission = get_perm_obj(permission_name, Datastream)
```

Comment:
This part can be done with the helper function that exists as well :)

`service.client.has_obj_perm(permission_name, ds, 'id')`
it will return True or False

Reply:
I'll change that.